File: Sample Apps /Java/SampleIndoorLocationReporting/app/src/main/java/com/mist/samplelocationbackground/initializer/MistSdkManager.java
package com.mist.samplelocationbackground.initializer;

import android.content.Context;
import android.widget.Toast;

import com.mist.android.BatteryUsage;
import com.mist.android.IndoorLocationCallback;
import com.mist.android.IndoorLocationManager;
import com.mist.android.external.config.LogLevel;
import com.mist.android.external.config.MistConfiguration;

import java.lang.ref.WeakReference;

/**
 * MistSdkInitializer This class provides utility function to start the Mist SDK.
 */
public class MistSdkManager {
    /**
     * Required by Mist SDK for initialization IndoorLocationManager IndoorLocationCallback
     * VirtualBeaconCallback
     */
    private IndoorLocationManager indoorLocationManager;
    private static WeakReference<Context> contextWeakReference;
    private String envType, orgSecret, orgId;
    private static MistSdkManager mistSdkManager;
    private MistConfiguration mistConfiguration;
    private IndoorLocationCallback indoorLocationCallback;

    private MistSdkManager() {
    }

    public static MistSdkManager getInstance(Context context) {
        contextWeakReference = new WeakReference<>(context);
        if (mistSdkManager == null) {
            mistSdkManager = new MistSdkManager();
        }
        return mistSdkManager;
    }

    public void init(String orgSecret, String orgId, IndoorLocationCallback indoorLocationCallback) {
        if (orgSecret != null && !orgSecret.isEmpty()) {
            this.orgSecret = orgSecret;
            this.orgId = orgId;
            this.indoorLocationCallback = indoorLocationCallback;
            this.envType = String.valueOf(orgSecret.charAt(0));
            this.mistConfiguration = new MistConfiguration(
                    contextWeakReference.get(),
                    this.orgSecret,
                    this.orgId,
                    "",
                    LogLevel.INFO,
                    true,
                    BatteryUsage.HIGH_BATTERY_USAGE_HIGH_ACCURACY
            );
        } else {
            Toast.makeText(contextWeakReference.get(), "Org Secret not present", Toast.LENGTH_SHORT).show();
        }
    }

    public synchronized void startMistSDK() {
        if (indoorLocationManager == null) {
            indoorLocationManager = IndoorLocationManager.INSTANCE;
            indoorLocationManager.start(mistConfiguration, indoorLocationCallback);
        } else {
            restartMistSDK();
        }
    }

    void stopMistSDK() {
        if (indoorLocationManager != null) {
            indoorLocationManager.stop();
        }
    }

    public void destroy() {
        if (indoorLocationManager != null) {
            indoorLocationManager.stop();
        }
    }

    private synchronized void restartMistSDK() {
        if (indoorLocationManager != null) {
            stopMistSDK();
            indoorLocationManager.start(mistConfiguration, indoorLocationCallback);
        }
    }
}

File: Sample Apps /Kotlin/SampleIndoorLocationReporting/app/src/main/java/com/example/sampleindoorlocationreporting/initializer/MistSdkManager.kt
package com.example.sampleindoorlocationreporting.initializer

import android.content.Context
import android.widget.Toast
import com.mist.android.BatteryUsage
import com.mist.android.IndoorLocationCallback
import com.mist.android.IndoorLocationManager
import com.mist.android.external.config.LogLevel
import com.mist.android.external.config.MistConfiguration
import java.lang.ref.WeakReference

/**
 * MistSdkInitializer This class provides utility function to start the Mist SDK.
 */

class MistSdkManager {
    /**
     * Required by Mist SDK for initialization IndoorLocationManager IndoorLocationCallback
     */
    private var indoorLocationManager : IndoorLocationManager? = null
    private var mistConfiguration: MistConfiguration? = null
    private lateinit var contextWeakReference : WeakReference<Context>
    private var envType: String?=null
    private var orgSecret : String?=null
    private var mistSdkManager : MistSdkManager? = null
    private var indoorLocationCallback: IndoorLocationCallback? = null


    fun getInstance(context: Context): MistSdkManager? {
        contextWeakReference = WeakReference<Context>(context)
        mistSdkManager?: run {
            mistSdkManager = MistSdkManager()
        }
        return mistSdkManager
    }

    fun init(orgSecret: String?, indoorLocationCallback: IndoorLocationCallback?, orgId: String) {
        this.indoorLocationCallback = indoorLocationCallback
        if (!orgSecret.isNullOrEmpty()) {
            this.orgSecret = orgSecret
            this.envType = orgSecret[0].toString()
            this.mistConfiguration = MistConfiguration(
                context = contextWeakReference.get()!!,
                token = orgSecret,
                enableLog = true,
                logLevel = LogLevel.INFO,
                batteryUsage = BatteryUsage.HIGH_BATTERY_USAGE_HIGH_ACCURACY,
                orgId = orgId
            )
        }
        else{
            Toast.makeText(contextWeakReference.get(), "Org Secret not present", Toast.LENGTH_SHORT).show();
        }
    }

    @Synchronized
    fun startMistSDK() {
        indoorLocationManager?.let {
            restartMistSDK()
        }?:run {
            indoorLocationManager = IndoorLocationManager
            mistConfiguration?.let {mistconfiguration->
                indoorLocationCallback?.let { indoorlocationcallback ->
                    indoorLocationManager?.start(
                        mistconfiguration,
                        indoorLocationCallback = indoorlocationcallback
                    )
                }
            }
        }
    }

    @Synchronized
    private fun stopMistSDK() {
        indoorLocationManager?.let{
            indoorLocationManager?.stop()
        }
    }

    fun destroy() {
        indoorLocationManager?.let{
            indoorLocationManager?.stop()
        }
    }

    @Synchronized
    fun restartMistSDK() {
        indoorLocationManager?.let {
            stopMistSDK()
            mistConfiguration?.let {mistconfiguration ->
                indoorLocationCallback?.let { indoorlocationcallback ->
                    indoorLocationManager?.start(
                        mistconfiguration,
                        indoorLocationCallback = indoorlocationcallback
                    )
                }
            }
        }
    }
}